src/pagePartials/strategy/vaults/VaultInfo.tsx

```diff
@@ -15,23 +15,24 @@ const Title = styled(InnerTitle)`
 export const VaultInfo: React.FC = withGenericSuspense(
   ({ ...restProps }) => {
     const { vaultAddress } = useVaultModalContext()
-    const { vaultBalance } = useGetUserVaultBalances(vaultAddress)
+    const { vaultBalances } = useGetUserVaultBalances(vaultAddress)
 
     return (
       <Rows {...restProps}>
         <Title>Vault balance</Title>
-        {vaultBalance?.length ? (
-          vaultBalance.map(({ agToken, balanceInDai, balanceRaw }, index) => {
+        {vaultBalances?.length ? (
+          vaultBalances.map(({ agToken, balanceInDai, balanceRaw, token }, index) => {
             // if index is multiple of 2, then it's a dark row
             const variant = index % 2 === 0 ? 'dark' : undefined
+
             return (
               <Row key={agToken.address} variant={variant}>
                 <RowKey>
                   <TokenIcon dimensions={18} symbol={agToken.symbol} />
                   {agToken.symbol}
                 </RowKey>
                 <EmphasizedRowValue>
-                  <Amount decimals={agToken.decimals} symbol="" value={balanceRaw} />
+                  <Amount decimals={token.decimals} symbol="" value={balanceRaw} />
```

Comment:
why token instead of `agToken`? is there a difference in the number of decimals between agUSDC and USDC for instance?

Reply:
related tokens don't return decimals from the token class.
you can check [here](https://github.com/Agave-DAO/frontend-v2/blob/main/src/config/agaveTokens.ts#L31) and [here](https://github.com/Agave-DAO/frontend-v2/blob/main/src/config/agaveTokens.ts#L31156)